clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImpl.java

```diff
@@ -1018,7 +1018,7 @@ private void acknowledgeBatchIfImplicitAcknowledgement(boolean calledOnPoll) {
             if (acknowledgementMode == AcknowledgementMode.UNKNOWN) {
                 // The first call to poll(Duration) moves into PENDING
                 acknowledgementMode = AcknowledgementMode.PENDING;
-            } else if (acknowledgementMode == AcknowledgementMode.PENDING) {
+            } else if (acknowledgementMode == AcknowledgementMode.PENDING && !currentFetch.isEmpty()) {
                 // The second call to poll(Duration) if PENDING moves into IMPLICIT
```

Comment:
I suggest a better comment such as `// If there are records to acknowledge and PENDING, moves into IMPLICIT`. It might not be the second call any more.